### PR TITLE
PRO-1811 Task: Modify BUIDLer for Prime

### PR DESCRIPTION
**Pull request by yashvyas19**
## Description

- Installed the Etherspot Prime SDK and Etherspot Transaction Kit as project dependencies.
- Fixed ESLint, formatting & typescript related issues at some places
- Introduced a new 'etherspotMode' property to enable mode selection.
- Depending on the 'etherspotMode' value, dynamically rendered common components and managed unified context providers for both modes.
       
## Motivation and Context

- https://linear.app/pillarproject/issue/PRO-1811/task-modify-buidler-for-prime
- https://linear.app/pillarproject/issue/PRO-1552/story-buidler-with-etherspot-prime

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by IAmKio — CHANGES_REQUESTED** [review]
Looking good so far just a clean-up operation needed

File: src/containers/Etherspot.tsx [hunk 1]
```diff
@@ -12,6 +12,9 @@ import {
 
 import { darkTheme, getTheme, Theme, ThemeType } from '../utils/theme';
 import { IDefaultTransactionBlock, ITransactionBlockType } from '../types/transactionBlock';
+import { EtherspotTransactionKit } from '@etherspot/transaction-kit';
+
+export const ETHERSPOT = 'etherspot';
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/containers/Etherspot.tsx]: Done

File: src/containers/Etherspot.tsx [hunk 2]
```diff
@@ -12,6 +12,9 @@ import {
 
 import { darkTheme, getTheme, Theme, ThemeType } from '../utils/theme';
 import { IDefaultTransactionBlock, ITransactionBlockType } from '../types/transactionBlock';
+import { EtherspotTransactionKit } from '@etherspot/transaction-kit';
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/containers/Etherspot.tsx]: Done

File: src/contexts/EtherspotContext.tsx [hunk 3]
```diff
@@ -17,6 +17,7 @@ import {
 } from '../providers/EtherspotContextProvider';
 import { Chain } from '../utils/chain';
 import { Theme } from '../utils/theme';
+import { PrimeSdk } from '@etherspot/prime-sdk';
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/contexts/EtherspotContext.tsx]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 4]
```diff
@@ -11,17 +11,28 @@ import {
...
+import {
+  useEtherspot,
+  useEtherspotAssets,
+  useEtherspotBalances,
+  useEtherspotNfts,
+  useEtherspotPrices,
+  useEtherspotTransactions,
+  useEtherspotUtils,
+  useWalletAddress,
+} from '@etherspot/transaction-kit';
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx:25]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 5]
```diff
@@ -57,25 +68,42 @@ const EtherspotContextProvider = ({
...
   const sessionStorage = etherspotSessionStorage ?? sessionStorageInstance;
 
-  const [accountAddress, setAccountAddress] = useState<string | null>(null);
-  const [providerAddress, setProviderAddress] = useState<string | null>(null);
+  // const [accountAddress, setAccountAddress] = useState<string | null>(
+  //   isEtherspotPrime(etherspotMode) ? primeAccountAddress! : null
+  // );
+  // const [providerAddress, setProviderAddress] = useState<string | null>(
+  //   isEtherspotPrime(etherspotMode) ? primeWalletAddress! : null
+  // );
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 6]
```diff
@@ -145,106 +173,116 @@ const EtherspotContextProvider = ({
...
+  //     computed = await (sdk as EtherspotSdk).computeContractAccount({ sync: true });
+  //     setIsConnecting(false);
+  //     return computed?.address;
+  //   } catch (e) {
+  //     console.log(e);
+  //     //
+  //   }
+
+  //   setIsConnecting(false);
+  // }, [sdk, isConnecting]);
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 7]
```diff
@@ -145,106 +173,116 @@ const EtherspotContextProvider = ({
...
+    //   return () => {
+    //     try {
+    //       if ((sdk as EtherspotSdk)?.state$?.closed) return;
+    //       // TODO: check why subscription in the above cannot be resubscribed
+    //       // sdk.state$.unsubscribe();
+    //     } catch (e) {
+    //       //
+    //     }
+    //   };
+    // }
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 8]
```diff
@@ -145,106 +173,116 @@ const EtherspotContextProvider = ({
...
-        if (MainnetIDs.includes(assetsChainId)) {
-          assets = await sdk.getTokenListTokens({
-            name: chainsToUseNewAssets.includes(assetsChainId) ? 'EtherspotPopularTokens' : 'PillarTokens',
-          });
-        } else {
-          assets = await sdk.getTokenListTokens();
+      if (isEtherspotPrime(etherspotMode)) {
+        try {
+          assets = await getAssets();
+        } catch (e) {}
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx]: > What happens if we cannot get assets from a user perspective?

If the assets are not successfully fetched due to an error, the assets array will remain empty

> Please add at least a console.error here or the logger equivalent

Done

File: src/providers/EtherspotContextProvider.tsx [hunk 9]
```diff
@@ -145,106 +173,116 @@ const EtherspotContextProvider = ({
...
+          CHAIN_ID.OPTIMISM,
+          CHAIN_ID.ARBITRUM,
+          CHAIN_ID.XDAI,
+          CHAIN_ID.BINANCE,
+          CHAIN_ID.ETHEREUM_MAINNET,
+          CHAIN_ID.POLYGON,
+          CHAIN_ID.OKTC,
+        ];
+
+        const MainnetIDs = Object.values(MAINNET_CHAIN_ID);
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 10]
```diff
@@ -145,106 +173,116 @@ const EtherspotContextProvider = ({
...
+        try {
+          if (MainnetIDs.includes(assetsChainId)) {
+            assets = await (sdk as EtherspotSdk).getTokenListTokens({
+              name: chainsToUseNewAssets.includes(assetsChainId) ? 'EtherspotPopularTokens' : 'PillarTokens',
+            });
+          } else {
+            assets = await (sdk as EtherspotSdk).getTokenListTokens();
+          }
+        } catch (e) {
+          //
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx:221]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 11]
```diff
@@ -439,15 +484,27 @@ const EtherspotContextProvider = ({
...
+        return prices.reduce<Record<string, number>>(
+          (currentRates, rate) => ({
+            ...currentRates,
+            [rate.address]: rate.usd,
+          }),
+          {}
+        );
+      }
     } catch (error) {
       //
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx:452]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 12]
```diff
@@ -456,12 +513,19 @@ const EtherspotContextProvider = ({
...
+          return rates.items[0].usd;
+        }
+      } else {
+        const price = await getPrice(ethers.constants.AddressZero, chainId);
+        if (price) {
+          return price.usd;
+        }
       }
     } catch (error) {
       //
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx:475]: Done

File: src/providers/EtherspotContextProvider.tsx [hunk 13]
```diff
@@ -580,10 +653,10 @@ const EtherspotContextProvider = ({
   const logout = useCallback(() => {
     sdkPerChain = {};
     setProvider(null);
-    setProviderAddress(null);
-    setAccountAddress(null);
+    // setProviderAddress(null);
+    // setAccountAddress(null);
```
[the remark this answers is not in the document]
Comment (yashvyas19) [reply src/providers/EtherspotContextProvider.tsx]: Done

**Review by IAmKio — APPROVED** [review]

**Pull request merged**